packages/sit-onyx/src/components/OnyxDataGrid/OnyxDataGrid.vue

```diff
@@ -0,0 +1,44 @@
+<script setup lang="ts" generic="TEntry extends DataGridEntry">
+import { ref, toRefs, watch, type Ref } from "vue";
+import {
+  OnyxDataGridRenderer,
+  type DataGridEntry,
+  type DataGridMetadata,
+  type DataGridRendererColumn,
+  type DataGridRendererRow,
+} from "../..";
+import { useTableFeatures } from "./features";
+import { useTableSorting } from "./features/sorting/sorting";
+
+const props = defineProps<{
+  /**
+   * The order of and which columns should be rendered.
+   */
+  columns: (keyof TEntry)[];
+  /**
+   * The data that should be used to fill the table.
+   */
+  data: TEntry[];
+}>();
+
+const withSorting = useTableSorting<TEntry>();
+
+const { watchSources, createRendererRows, createRendererColumns } = useTableFeatures([withSorting]);
+
+const renderCols: Ref<DataGridRendererColumn<TEntry, object>[]> = ref([]);
+const renderRows: Ref<DataGridRendererRow<TEntry, DataGridMetadata>[]> = ref([]);
+
+const { columns, data } = toRefs(props);
+
+watch(
+  [columns, data, ...watchSources],
+  ([newColumns, newData]) => {
+    renderCols.value = createRendererColumns(newColumns);
+    renderRows.value = createRendererRows(newData, newColumns);
+  },
+  { immediate: true },
```

Comment:
```suggestion
  { immediate: true },
```

Do we need a deep watcher here?

Reply:
Afaik, deep watchers are only used for function watchers: `() => watchedData`. We only use `Ref`s which are already tracking deep access.